src/basic/in-addr-util.c

```diff
@@ -179,44 +179,89 @@ int in_addr_prefix_next(int family, union in_addr_union *u, unsigned prefixlen)
         /* Increases the network part of an address by one. Returns
          * positive it that succeeds, or 0 if this overflows. */
 
+        return in_addr_prefix_nth(family, u, prefixlen, 1);
+}
+
+/*
+ * Calculates the nth prefix of size prefixlen starting from the address demoted by u.
+ *
+ * On success 1 will be returned and the calculated prefix written to u. In
+ * case the calculation can not be performed (invalid prefix length, overflows
+ * would occur) 0  is returned.
+ * If the address family given isn't supported -EAFNOSUPPORT will be returned.
+ *
+ *
+ * Examples:
+ *   - in_addr_prefix_nth(AF_INET, 192.168.0.0, 24, 2), returns 1, writes 192.168.2.0 to u
+ *   - in_addr_prefix_nth(AF_INET, 255.255.255.0, 24, 1), returns 0, no data written
```

Comment:
I think we should return 0 on succes and some error like -EINVAL or -ERANGE or so on failure. per our coding style we try to stick to negative errno-style error numbers for indicating failure, so let's do so here too

Reply:
I agree. I just kept that pattern for consistency as `in_addr_prefix_next` and `in_addr_random_prefix` did it like that. Changing success from `1` to `0` would probably only make sense if it would be changed in all of these cases.

Looking further in the file it seems like that pattern has been adopted by most of the functions in the file. It probably originates from the equality-family of functions where returning `1` is actually desirable (`if (something_eq(foo))` is just more readable than `if (something_eq(foo) == 0)` or `!something_eq(foo)`).

Even thought I think it might be a bit large of an (additional) scope for this PR I am happy to attempt it as part of this if, you insist. :)